solidity.pegjs

```diff
@@ -1424,8 +1456,18 @@ SourceElements
     }
 
 SourceElement
-  = AssignmentExpression __ EOS
-  / DeclarativeExpression __ EOS
+  = exp:AssignmentExpression  __ EOS &{
+      return (
+        exp.type === 'AssignmentExpression' && 
+        exp.left.type === 'DeclarativeExpression' && 
+        exp.left.storage_location === null
+      )
+    }{

```

Comment:
Not necessarily. Just a single rule matching the official specification:
https://github.com/federicobond/solidity-antlr4/blob/master/Solidity.g4#L57